### PR TITLE
fix(explore): make horizontal scroll appears in data panel

### LINKED ISSUE #12891: Horizontal scroll not working in explore data panel
### Expected results
In explore, if we resize the screen to a smaller width the horizontal scroll of the data panel appears and I can view all columns.

### Actual results
In explore, if we resize the screen to a smaller width the horizontal scroll of the data panel disappears and it's not possible to view all columns. Also as you can see in the gif, a row is appearing bellow the pagination component.

#### Screenshots
screen

#### How to reproduce the bug
1. Go to Explore
2. Resize width
3. Check data panel scroll

### Environment
- superset version:  `master`
- python version: `2.7.16`
- node.js version: `15.5.1`

### Checklist
- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by stephenLYZ**
### SUMMARY

Because the height of the tabpanel is higher than the height of the outer parent, the horizontal scroll bar is squeezed below.
Here need to minus some height to show horizontal scroll bar. Also the pagination's padding-bottom should be set to zero which causes a row is appearing bellow the component.

#### Before
image
#### After
Kapture

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #12891 
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by junlincc** [comment]
@villebro @rusackas 
Hey Ville and Evan, Stephen is a developer that me and Yongjie "interviewed" last week, he picked out a task from bash! and opened this PR as requested. please review. :)

**Comment by junlincc** [comment]
@kgabryje  ^ you wanna take a look and let us know your thoughts? thanks!

**Comment by zhaoyongjie** [comment]
@stephenLYZ could you fix the lint in this PR?

**Comment by stephenLYZ** [comment]
@zhaoyongjie done

File: superset-frontend/src/explore/components/DataTablesPane.tsx [hunk 1]
```diff
@@ -271,7 +271,7 @@ export const DataTablesPane = ({
               height: 100%;
 
               .ant-collapse-content {
-                height: 100%;
+                height: calc(100% - 33px);
```
Comment (kgabryje) [inline superset-frontend/src/explore/components/DataTablesPane.tsx:274]: Can we use `theme.gridUnit * 8` instead of `33px`?

**Comment by kgabryje** [comment]
1 nit, other than that LGTM. 
@junlincc What do you think about displaying horizontal scrollbar permanently so that user knows that there are more columns? It should work on Chrome and Safari (not Firefox though).

File: superset-frontend/src/explore/components/DataTablesPane.tsx (hunk 1 above)
Comment (stephenLYZ) [reply superset-frontend/src/explore/components/DataTablesPane.tsx:274]: ye！It's better than hardcode

**Comment by stephenLYZ** [comment]
> 1 nit, other than that LGTM.
> @junlincc What do you think about displaying horizontal scrollbar permanently so that user knows that there are more columns? It should work on Chrome and Safari (not Firefox though).

How about show all columns like chart table which is responsive?

File: superset-frontend/src/explore/components/DataTablesPane.tsx (hunk 1 above)
Comment (stephenLYZ) [reply superset-frontend/src/explore/components/DataTablesPane.tsx]: done, please review, thanks!

Comment (kgabryje) [reply superset-frontend/src/explore/components/DataTablesPane.tsx]: LGTM

**Review by villebro — APPROVED** [review]
LGTM!

**Review by junlin-qa — APPROVED** [review]
Responsiveness is probably better in this case. lets merge! 🎉 thank you stephen!!

**Pull request merged**